src/state/store.js

```diff
@@ -1,25 +1,91 @@
-import { derived } from 'svelte/store';
+import { derived, get } from 'svelte/store';
 
 import { createStore } from '../utils/create-store';
 
 // FIXME: take care of this dependency cycle.
 import { probeSet } from './telemetry-search'; // eslint-disable-line import/no-cycle
 
-import CONFIG from '../config/firefox-desktop';
+import sharedConfig from '../config/shared';
+import productConfig from '../config/products';
 
-// TODO: move this to the new config.js when 'product' is added.
-const DEFAULT_PROBE_PROCESS = 'content';
+export function getFromQueryString(fieldKey, isMulti = false) {
+  const params = new URLSearchParams(window.location.search);
+  const value = params.get(fieldKey);
+  if (isMulti) {
+    return JSON.parse(value);
+  }
+  return value;
+}
+
+function getDefaultState(
+  { basedOnQueryParams } = { basedOnQueryParams: false }
+) {
+  const state = {};
+
+  state.auth = {
+    isAuthenticated: false,
+    token: undefined,
+  };
+
+  // FIXME: applicationStatus or dashboardMode, not both.
+  state.applicationStatus = 'INITIALIZING';
+  state.dashboardMode = {};
+
+  state.product = window.location.pathname.split('/')[1] || 'firefox';
+  state.probeName = '';
+  state.recordedInProcesses = []; // Provided by the API. List of processes this probe was recording in.
+  state.reference = '';
+  state.route = {};
+
+  // Shared config
+  Object.entries(sharedConfig).forEach(([key, { isMulti, defaultValue }]) => {
+    if (basedOnQueryParams) {
+      state[key] = getFromQueryString(key, isMulti) || defaultValue;
+    } else {
+      state[key] = defaultValue;
+    }
+  });
+
+  // Product config
+  state.productDimensions = {};
+  Object.entries(productConfig[state.product].dimensions).forEach(
+    ([key, { defaultValue }]) => {
+      if (basedOnQueryParams) {
+        state.productDimensions[key] = getFromQueryString(key) || defaultValue;
+      } else {
+        state.productDimensions[key] = defaultValue;
+      }
+    }
+  );
+
+  return state;
+}
+
+export const store = createStore(getDefaultState());
+
+store.reset = () => {
+  store.reinitialize({
+    auth: store.getState().auth,
+    probeName: '',
+  });
+};
+
+const active = {
+  get productConfig() {
+    return productConfig[get(store).product];
+  },
+};
```

Comment:
This allows us to do `active.productConfig.dimensions.[whatever]`. Another pattern I considered was adding a `get` method to the imported `productConfig` object, which would allow us to do `productConfig.currentProduct.dimensions.[whatever]`, which would look like this:

```javascript
Object.defineProperty(productConfig, 'currentProduct', {
  get() {
    return productConfig[get(store).product];
  },
});

const whatever = productConfig.currentProduct.dimensions.[whatever];
```

I'd be interested to hear which you think is more readable.

Reply:
I think the latter is a bit more readable to me b/c it's tied a bit more obviously to the config so I know where to look for the value if I see `productConfig.currentProduct` elsewhere. Also aren't we going to need `productConfig.currentProduct` elsewhere, such as in the dimension selector region? Is the goal to have other potential active things in `active`? If so, then I think this is a good approach, but I can't think of another case that we'd need this. My pref would be to go for `productConfig.currentProduct` with this in mind, but this is a weak position – either works for me.

Reply:
Sounds good. We may need `currentProduct` elsewhere, but I think *store.js* is mainly where things get weird: we need to read from the store itself to determine what `currentProduct` is.

I could add a `currentProduct` property to *config/products.js*. That would create a dependency cycle. I'm not certain that the dependency cycle would cause problems, but I try to avoid them in general.

Reply:
I opted for something a little easier with less magic. May have been overthinking it.